src/rgb_led.rs

```diff
@@ -76,3 +78,27 @@ impl<F: FullDuplex<u8>> LedStrip<F> {
         }
     }
 }
+
+#[allow(dead_code)]
+pub struct Pulser {
+    instant: Instant,
+    interval_ticks: f32,
+}
+
+impl Pulser {
+    pub fn new(interval_ms: u32, timer: &MonoTimer) -> Self {
+        let instant = timer.now();
+        let interval_ticks = timer.frequency().0 as f32 * (interval_ms as f32 / 1000.0);
+
+        Self { instant, interval_ticks }
+    }
+
+    #[allow(dead_code)]
+    pub fn intensity(&self) -> f32 {
+        let intervals = self.instant.elapsed() as f32 / self.interval_ticks;
+        let pulse = (libm::sinf(intervals) + 1.0) * 0.5;
+        let skip_one = if libm::sinf((intervals + PI / 2.0) / 2.0) >= 0.0 { 1.0 } else { 0.0 };
+
+        pulse * skip_one
+    }
+}
```

Comment:
This is much better! I changed the pulse interval to 700ms and it feels pretty close to the old Mac sleep LED.

I do notice a slight "skip" on occasion when it's increasing in brightness, not yet sure what that's about. Possibly when the mono timer's elapsed time reaches `u32::max`?

Reply:
So far I can confirm that it happens after 1 minute, 30 seconds into the firmware starting up, which lines up with a 48MHz clock.

(2^32) / 48,000,000 = 89.48 ~= 1 minute, 29 seconds

Reply:
@bschwind good catch :) I don't know how the u32 counter wraps, but do you mind testing my latest commit which tries to work it around?

Reply:
They seem to wrap to zero when overflowing, so maybe my code works:
https://stackoverflow.com/questions/53434504/stm32f411-dwt-cyccnt-counter-width